### DISCUSSION TITLE
My first text(Tweeter) mining Python code ERROR

### DISCUSSION THREAD

**Discussion by BinTEPO** [community/community#21393] [Other Feature Feedback, Questions, & Ideas]
<p><strong>Can someone help please, I am trying to extract some tweets and then perform some basic analysis. I am encountering an error and i suspect its around the below section of the code:</strong></p>
<p><strong>I defined a function called word_in_text(word, text). This function return True if a word is found in text, otherwise it returns False</strong></p>
<p>def word_in_text(word, text):<br>
    word = word.lower()<br>
    text = [item.lower() for item in tweet]</p>
<p>    match = re.search(word, text)</p>
<p><strong>I get the following ERROR</strong></p>
<p>File “*:**\Anaconda3\lib\re.py”, line 183, in search<br>
return _compile(pattern, flags).search(string)</p>
<p>TypeError: expected string or bytes-like object</p>
<p>Thank you</p>

**Comment by mpboom** [comment, marked as answer]
<p>Hi and welcome <a class="mention" href="/bintepo">@bintepo</a> ,</p>
<p>To match using the package that provides regular expressions, you must first craft a regular expression. Here are some handy tools:</p>
<ul>
<li>An introduction (with examples in Python!): <a href="https://www.oreilly.com/ideas/an-introduction-to-regular-expressions" rel="nofollow noopener">https://www.oreilly.com/ideas/an-introduction-to-regular-expressions</a>
</li>
<li>Playing around with them: <a href="https://regexr.com/" rel="nofollow noopener">https://regexr.com/</a>
</li>
<li>Implementing them in Python: <a href="https://www.w3schools.com/python/python_regex.asp" rel="nofollow noopener">https://www.w3schools.com/python/python_regex.asp</a>
</li>
</ul>
<p>After catching up on that and getting to know regular expressions you should be able to re-write the function to let it perform how you’d like to.</p>
<p>I know it’s quite a bit to learn but it’s pretty fun and if you’re performing analysis on Tweets regular expressions will come in handy!</p>
<p>As always, just post back if you encounter any problems.</p>
